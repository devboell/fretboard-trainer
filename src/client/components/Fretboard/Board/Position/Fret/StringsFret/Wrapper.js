import styled, { css } from 'styled-components'

const highlightCSS = ({ theme }) => {
  const radiusPerc = 80
  const radius = Math.floor((theme.stringHeight / 100) * radiusPerc)

  return css`
    width: ${radius}px;
    height: ${radius}px;
    border-radius: 50%;
    background-color: ${theme.highlight};
  `
}


const Wrapper = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;

  ${props => (props.isHighlighted
    ? highlightCSS(props)
    : css`width: 100%;`)
} 
`

export default Wrapper
