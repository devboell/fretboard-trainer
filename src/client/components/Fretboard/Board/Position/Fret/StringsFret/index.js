import React from 'react'
import pt from 'prop-types'
import StringsContent from '../StringsContent'
import Wrapper from './Wrapper'


const StringsFret = ({
  note, showNotes, isHighlighted,
}) =>
  <Wrapper {...{ isHighlighted }}>
    {showNotes &&
      <StringsContent {...{ note, isHighlighted }} />
    }
  </Wrapper>

StringsFret.propTypes = {
  note: pt.string.isRequired,
  showNotes: pt.bool.isRequired,
  isHighlighted: pt.bool.isRequired,
}

export default StringsFret
