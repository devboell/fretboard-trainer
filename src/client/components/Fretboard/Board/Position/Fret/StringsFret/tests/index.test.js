import React from 'react'

import StringsFret from '../index'

describe('StringsFret component', () => {
  it('makes snapshot', () => {
    const wrapper = shallow(<StringsFret note="E2" showNotes />)
    expect(wrapper).toMatchSnapshot()
  })

  it('makes snapshot, showNotes={false}', () => {
    const wrapper = shallow(<StringsFret note="E2" showNotes={false} />)
    expect(wrapper).toMatchSnapshot()
  })
})
