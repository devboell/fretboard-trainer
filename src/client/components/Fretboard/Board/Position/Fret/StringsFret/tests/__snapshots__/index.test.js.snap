// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StringsFret component makes snapshot 1`] = `
<styled.div
  isHighlighted={false}
>
  <StringsContent
    isHighlighted={false}
    note="E2"
  />
</styled.div>
`;

exports[`StringsFret component makes snapshot, showNotes={false} 1`] = `
<styled.div
  isHighlighted={false}
/>
`;
